deliverables/M2-Group2.md

```diff
@@ -0,0 +1,67 @@
+
+**6- API System Requirements:** The first thing we did before implementation phase was to write system requirements. We decided to provide basic investments and parity functionalities, and wrote down the requirements accordingly. Then, we used this to share the responsibilities between us. We created 10 requirement items and each member was supposed to implement one of them. We have done this to provide equal opportunity of implementation to everyone. 
+
+**7- API Design Diagram:** We have use case diagram, class diagram and sequence diagrams. The class diagrams was almost a subset of the class diagram we did before, with minor modifications. The use case diagram and sequence diagram was also similar to the previous deliverables we have had. All in all, as usual, preparing these diagrams helped us predict and build the workflow of the API: models, methods etc. 
+
+**8- Collaboration Workflow Documentation:** This was one of the best things we did as a team. We prepared issue templates and pull request templates. In the pull request templates, we have 2 main things, first is making sure that the author has done everything that she was supposed to do, such as adding 2 reviewers, self-reviewing and using the format(PEP8) we agreed to use, and secondly describing what is the nature and philosophy of the implementation, to let the reviewers grasp the material easier. We did the same thing with the issue templates, with the exact same purpose. Using these templates let us be on the same page using pull requests and issues system. Overall this is one of the things we consider the highlight of our group.
+
+**9- API Code:** We used Django to implement the backend of the app, and used javascript jquery to implement the frontend. We have basically 2 family of functionalities, one is investments and the other family is parity. We provide creating, deleting, listing investments and profit calculation. We also provide listing history of parities, latest parity values. We also have register and login functionalities, secured by JWT Authentication. One thing we did good is the cron job of our backend, the job basically sends request to the API we have used(https://exchangeratesapi.io/) and updates the database accordingly. When the system is initialized, we query the API using our own-written django command to fetch the history of selected parities, then cronjob constantly updates the database with new values. Hence, the parity latest and historic endpoints can provide the information fast and robust, instead of querying another API when the request sent to us. We may even directly use these systems once we start next term, since these parts are working and solid right now.
+
```

Comment:
Nice work 💯. Can we also mention Travis and testing somewhere here? Maybe **8** ?

Reply:
thanks, just added 👌